dogfood/Dockerfile

```
@@ -312,8 +312,30 @@ COPY --from=go /tmp/bin /usr/local/bin
 COPY --from=rust-utils /tmp/bin /usr/local/bin
 COPY --from=proto /tmp/bin /usr/local/bin
 
+# Configure Nix without sandboxing
+# - https://github.com/NixOS/nix/issues/2636#issuecomment-455302745
+# - https://nixos.org/manual/nix/stable/installation/multi-user.html#setting-up-the-build-users
+RUN addgroup --system nixbld \
+  && adduser coder nixbld \
+  && for i in $(seq 1 30); do useradd -ms /bin/bash nixbld$i &&  adduser nixbld$i nixbld; done \
+  && mkdir -m 0755 /nix && chown coder:coder /nix \
+  && mkdir -p /etc/nix && echo 'sandbox = false' > /etc/nix/nix.conf
+
 USER coder
 
+# Install Nix
+ARG NIX_VERSION=2.3.15
+RUN curl --silent --show-error --location \
+  "https://releases.nixos.org/nix/nix-${NIX_VERSION}/nix-${NIX_VERSION}-x86_64-linux.tar.xz" \
+  -o "nix-${NIX_VERSION}-x86_64-linux.tar.xz"
+
+RUN tar -xf "nix-${NIX_VERSION}-x86_64-linux.tar.xz" \
+  && bash "nix-${NIX_VERSION}-x86_64-linux/install" "--no-daemon" \
+  && rm -rf "nix-${NIX_VERSION}-x86_64-linux.tar.xz"
```

I think we'll still be mounting an empty `/nix` over the top via the `main.tf` volume, so I'd suggest we rewrite this like so (also reducing a layer in the final image):

```suggestion
RUN cd /opt \
  && curl --silent --show-error --location \
  "https://releases.nixos.org/nix/nix-${NIX_VERSION}/nix-${NIX_VERSION}-x86_64-linux.tar.xz" \
  -o "nix-${NIX_VERSION}-x86_64-linux.tar.xz" \
  && tar -xf "nix-${NIX_VERSION}-x86_64-linux.tar.xz" \
  && ln -s "nix-${NIX_VERSION}-x86_64-linux" nix \
  && rm -rf "nix-${NIX_VERSION}-x86_64-linux.tar.xz"
```

And in the agent startup:

```sh
bash /opt/nix/install --no-daemon
```

I'm assuming it'll be no-op/upgrade if installed on top of an existing one?